### PR TITLE
Update to Ruby 3.0.2

### LINKED ISSUE #40: Tests for Word Count pass locally with Ruby 2.6 but fail on Exercism website because test runner is using Ruby 2.5.9
In the Word Count exercise, I'm using `to_h` with a block, which only becomes available in Ruby 2.6. 
https://rubyreferences.github.io/rubychanges/2.6.html#to_h-with-a-block

The error shows for this line in my code: 
`words.to_h {|word| [word, words.count(word)] }`

> TypeError: wrong element type String at 0 (expected array)
> 
> Traceback (most recent call first):
>     Line 10:in 'to_h'
>     Line 10:in 'word_count'

I found the Ruby version by adding this in the code editor:
`"ruby #{ RUBY_VERSION }p#{ RUBY_PATCHLEVEL }".frog`
> NoMethodError: undefined method `frog' for "ruby 2.5.9p229":String

[After that, I discovered there's `debug` method! 😄 ]

I came here from: https://exercism.org/docs/using/solving-exercises/tests-pass-locally 
although I see that the code in this repo already has a .ruby-version of ruby-2.6.6. 

I also see that https://github.com/exercism/ruby says
> You'll need a recent (2.6+) version of Ruby

So I assume at some point the test runner on the website will be updated to 2.6?
Not sure if this is the right place to report this. Thanks! 😊

### PR DESCRIPTION
**Pull request by katrinleinweber**
Thanks to @iHiD's advice in #41, I think this resolves #40 in a more complete way:

1. Documents the advice
1. `docker build .` succeeds
1. `bundle exec rake test` as well: `12 runs, 21 assertions, 0 failures, 0 errors, 0 skips`

---

This PR also bumps gems to newer versions:

- `mandate`, [as advised](https://github.com/exercism/ruby-test-runner/pull/41#issuecomment-961822998)
-  `json`,  to resolve `ArgumentError: wrong number of arguments (given 2, expected 1)` in `TestRunnerTest#…`
- `parser`, to stop `warning: parser/current is loading parser/ruby27, which recognizes 2.7.x-compliant syntax, but you are running 3.0.2.`

---

Additionally, a blanket `bundle update` could also be added here.

### CODE REVIEW

**Comment by kotp** [comment]
That is strange, because the [README from setup-ruby][1] says: 

> 1.9.3, 2.0.0, 2.1.9, 2.2, all versions from 2.3.0 until 3.0.2, head, debug, mingw, mswin

[1]: https://github.com/ruby/setup-ruby

**Comment by kotp** [comment]
Don't forget to specify the Ruby version in the Gemfile as well.  It might be the reason for the failure(??).

File: test/test_runner_test.rb [hunk 1]
```diff
@@ -124,7 +124,7 @@ def test_syntax_errors_in_code
         end,A stray comma
            ^
       SYNTAX_ERRORS
-      assert_equal expected.strip, actual['message']
+      assert_equal expected, actual['message']
```
Comment (iHiD) [inline test/test_runner_test.rb:127]: What's the reason for these changes?

File: README.md [hunk 2]
```diff
@@ -17,10 +17,15 @@ For example:
...
+bundle config set --local with 'test'
 bundle install
 ```
 
-Then, run the following command to run the tests:
+Then, in [test/test_helper.rb](test/test_helper.rb),
+comment the "Production command" `system(…)` out
+and uncomment either of the "Testing commands".
+
+Lastly, run the following command to run the tests:
```
Comment (iHiD) [inline README.md:28]: Nice. Thanks.

File: README.md [hunk 3]
```diff
@@ -17,10 +17,15 @@ For example:
 Before running the tests, first install the dependencies:
 
 ```bash
+bundle config set --local with 'test'
```
Comment (iHiD) [inline README.md:20]: What does this do?

**Comment by iHiD** [comment]
@katrinleinweber Great. Thanks! I'll release a proper mandate version too. You ok to try and get the rubocop thing fixed or want me to look at it?

File: test/test_runner_test.rb (hunk 1 above)
Comment (katrinleinweber) [reply test/test_runner_test.rb:127]: Without that, 2 tests fail:

```
|   1) Failure:
| TestRunnerTest#test_syntax_errors_in_tests [/Users/katrinleinweber/Exercism.io/ruby-test-runner/test/test_runner_test.rb:111]:
| Expected: "Line 3: syntax error, unexpected ',', expecting end-of-input\n,'This is meant to be a syntax...\n^"
|   Actual: "Line 3: syntax error, unexpected ',', expecting end-of-input\n,'This is meant to be a syntax...\n^\n"
| 
|   2) Failure:
| TestRunnerTest#test_syntax_errors_in_code [/Users/katrinleinweber/Exercism.io/ruby-test-runner/test/test_runner_test.rb:127]:
| Expected: "Line 2: syntax error, unexpected ',', expecting end-of-input\nend,A stray comma\n   ^"
|   Actual: "Line 2: syntax error, unexpected ',', expecting end-of-input\nend,A stray comma\n   ^\n"
| 
| 12 runs, 19 assertions, 2 failures, 0 errors, 0 skips
```

I added an explanation to the description of https://github.com/exercism/ruby-test-runner/pull/47/commits/7cdb9402eb324baefa0fbd2e1aca6799481c8370 ;-)

File: README.md (hunk 3 above)
Comment (katrinleinweber) [reply README.md:20]: `bundle install --with=test` produces that advice, so I thought it would be useful to add it to the instructions directly, because testing will require the installation of [that gem group](https://github.com/exercism/ruby-test-runner/blob/9345867127a29b3aebcdb4b9d077a4b5f01bb9a1/Gemfile#L16-L23).

File: README.md (hunk 2 above)
Comment (katrinleinweber) [reply README.md:28]: You're welcome :-)

**Comment by katrinleinweber** [comment]
> … You ok to try and get the rubocop thing fixed or want me to look at it?

I though https://github.com/exercism/ruby-test-runner/pull/47/commits/a62c037b508bf368927355a80e91d2371328f079 would do that, but I forgot he second occurrence m-( Fixed now & force-pushed.

**Comment by iHiD** [comment]
I can't reopen - it won't let me. Have you deleted the branch maybe?

**Comment by Steffan153** [comment]
This helpful? https://gist.github.com/robertpainsi/2c42c15f1ce6dab03a0675348edd4e2c

**Comment by katrinleinweber** [comment]
@iHiD I'm fine with either option (☝️ or a new PR). I didn't delete `update-to-ruby-3.0.2` and pushed [a 2nd copy](https://github.com/katrinleinweber/ruby-test-runner/branches/all?query=3.0.2) just now.

Thanks for the gist, @Steffan153 :-)

**Comment by iHiD** [comment]
Could you put another PR up then pls :)

**Comment by katrinleinweber** [comment]
Sure thing: https://github.com/exercism/ruby-test-runner/pull/48 :-)
